perfkitbenchmarker/linux_benchmarks/object_storage_service_benchmark.py

```diff
@@ -293,9 +326,145 @@ def _MakeSwiftCommandPrefix(auth_url, tenant_name, username, password):
   return ' '.join(options)
 
 
+def _ParseMultiStreamResults(raw_result, metadata=None):
+  """Read results from the api_multistream worker process.
+
+  Args:
+    raw_result: string. The stdout of the worker process.
+    metadata: dict. Base sample metadata
+
+  Returns:
+    A list of sample.Sample objects.
+  """
+
+  if metadata is None:
+    metadata = {}
+
+  records = json.loads(raw_result)
+  results = []
+
+  for record in records:
+    metric = 'Multi-stream %s KB %s' % (record['size'], record['operation'])
+    md = metadata.copy()
+    md['stream_num'] = record['stream_num']
+    md['object_size_b'] = record['size']
+
+    results.append(sample.Sample(metric,
+                                 record['latency'],
+                                 'sec',
+                                 md,
+                                 record['start_time']))
+
+  return results
+
+
+def _StringToBytes(string):
+  """Convert an object size, represented as a string, to bytes.
+
+  Args:
+    string: the object size, as a string with a quantity and a unit.
+
+  Returns:
+    an integer. The number of bytes in the size.
+
+  Raises:
+    ValueError, if either the string does not represent an object size
+    or if the size does not contain an integer number of bytes.
+  """
+
+  try:
+    quantity = perfkitbenchmarker.UNIT_REGISTRY.parse_expression(string)
+  except Exception:
+    # Catching all exceptions is ugly, but we don't know what sort of
+    # exception pint might throw, and we want to turn any of them into
+    # ValueError.
+    raise ValueError("Couldn't parse object size %s" % string)
+
+  try:
+    bytes = quantity.m_as(perfkitbenchmarker.UNIT_REGISTRY.bytes)
+  except pint.DimensionalityError:
+    raise ValueError("Quantity %s is not an object size" % string)
+
+  if bytes != int(bytes):
+    raise ValueError("Size %s has a non-integer number (%s) of bytes!" %
+                     (string, bytes))
+
+  if bytes < 0:
```

Comment:
According to http://stackoverflow.com/questions/4581842/python-integer-ranges , and verified by my own experiments, Python ints are automatically promoted to longs when they overflow, rather than wrapping around. Also, the `int` conversion function will return a long rather than raising an exception if its argument (I tested strings and floats) doesn't fit in an int. Therefore, I think this code will work correctly, except that `StringToBytes` may sometimes return a long when it normally returns an int. This could only be an issue when the user specifies an object with more than 2^63 bytes, which is 8 exabytes. I don't think we need to worry about that case.